src/pages/resultsView/coExpression/CoExpressionViz.tsx

```diff
@@ -99,6 +107,19 @@ export default class CoExpressionViz extends React.Component<ICoExpressionVizPro
         });
     }
 
+    readonly coExpressionsWithQValues = remoteData<(CoExpression & { qValue:number})[]>({
+        await:()=>[this.coExpressionDataPromise],
+        invoke:()=>{
+            const coexpressions = this.coExpressionDataPromise.result!;
+            const sortedByPvalue = _.sortBy(coexpressions, c=>c.pValue);
+            const qValues = calculateQValues(sortedByPvalue.map(c=>c.pValue));
+            for (let i=0; i<qValues.length; i++) {
```

Comment:
curious why you like to use for loops rather than forEach?   the only reason i see is optimization

Reply:
which is a fine reason if there is actually a need